src/test/java/com/booster/server/NoteControllerTest.java

```diff
@@ -18,30 +26,69 @@
 @AutoConfigureMockMvc
 class NoteControllerTest {
 
+    final String firstNoteContent = "My first note";
+    final String secondNoteContent = "Read 10 pages";
+    final String thirdNoteContent = "Practice piano";
+
     @Autowired
     MockMvc mockMvc;
     @Autowired
     ObjectMapper objectMapper;
     @Autowired
     NoteRepository noteRepository;
 
+    @BeforeEach
+    void beforeEach() {
+        noteRepository.deleteAll();
+    }
+
     @Test
     void createNewNote() throws Exception {
         assertThat(noteRepository.findAll()).isEmpty();
 
-        String content = "My first note";
-
         mockMvc.perform(MockMvcRequestBuilders.post("/note")
                         .contentType(MediaType.APPLICATION_JSON_VALUE)
-                        .content(objectMapper.writeValueAsString(new CreateNoteInput().setContent(content))))
+                        .content(objectMapper.writeValueAsString(new CreateNoteInput().setContent(firstNoteContent))))
                 .andDo(print())
                 .andExpect(status().isCreated())
-                .andExpect(jsonPath("$.content").value(content));
+                .andExpect(jsonPath("$.content").value(firstNoteContent));
 
         assertThat(noteRepository.findAll())
                 .hasSize(1)
-                .extracting("content")
-                .containsExactly(content);
+                .extracting("content", "lastSeenAt")
+                .containsExactly(Tuple.tuple(firstNoteContent, LocalDateTime.of(LocalDate.EPOCH, LocalTime.MIN)));
+    }
+
+    @Test
+    void correctUpdateOfLastSeenAtPropertyOfNote() throws Exception {
+        assertThat(noteRepository.findAll()).isEmpty();
+
+        Long id1 = noteRepository.save(new NoteEntity()
+                        .setContent(firstNoteContent))
+                .getId();
+        Long id2 = noteRepository.save(new NoteEntity()
+                        .setContent(secondNoteContent))
+                .getId();
+        Long id3 = noteRepository.save(new NoteEntity()
+                        .setContent(thirdNoteContent))
+                .getId();
+
+        LocalDateTime lastSeenAt = LocalDateTime.now();
+        List<Long> ids = List.of(id1, id3);
+        mockMvc.perform(MockMvcRequestBuilders.patch("/note")
+                        .contentType(MediaType.APPLICATION_JSON_VALUE)
+                        .content(objectMapper.writeValueAsString(new UpdateLastSeenAtInput()
+                                .setIds(ids)
+                                .setLastSeenAt(lastSeenAt))))
+                .andDo(print())
+                .andExpect(status().isOk());
+
+        assertTrue(noteRepository.findAllById(ids)
+                .stream()
+                .allMatch(n -> n.getLastSeenAt().equals(lastSeenAt)));
+        assertThat(noteRepository.findById(id2)
+                .map(NoteEntity::getLastSeenAt).get())
+                .isEqualTo(LocalDateTime.of(LocalDate.EPOCH, LocalTime.MIN));
```

Comment:
We can get rid of the `get` method on `Optional` if we use the `contains` instead of `isEqualTo`.